server/middleware/populateCurrentUser.ts

```diff
@@ -1,14 +1,27 @@
 import { RequestHandler } from 'express'
 import logger from '../../logger'
+
 import UserService, { DeliusAccountMissingStaffDetailsError } from '../services/userService'
+import inMemoryStore from '../inMemoryStore'
 
 export default function populateCurrentUser(userService: UserService): RequestHandler {
   return async (req, res, next) => {
     try {
       if (res.locals.user) {
-        const user = req.session.user || (await userService.getActingUser(res.locals.user.token))
+        let { user } = req.session
+
+        if (!user || user.version !== inMemoryStore.users[user.id]) {
+          const currentVersion = user?.version
+          user = await userService.getActingUser(res.locals.user.token)
+          if (user) {
+            logger.info(
+              `Updated user from API${currentVersion !== user.version ? ` (version: ${currentVersion} -> ${user.version})` : ''}`,
+            )
+          }
+        }
+
         req.session.user = user
-        res.locals.user = { ...user, ...res.locals.user }
+        res.locals.user = { ...res.locals.user, ...user }
```

Comment:
Shouldn't these two lines be inside the user version changed test block? I guess it would just be overwriting the session and local user with the same data but it seems a bit pointless.

Reply:
I guess that's correct for assigning `user` to `req.session.user`, yes!

But for the assignment to locals, if we're populating the user details from what is available in the session, we still want this data to be added to `res.locals` so it can be used in templates.